src/OrchardCore.Modules/OrchardCore.PublishLater/Services/ScheduledPublishingBackgroundTask.cs

```diff
@@ -30,7 +29,6 @@ public async Task DoWorkAsync(IServiceProvider serviceProvider, CancellationToke
             var itemsToPublish = await serviceProvider
                 .GetRequiredService<ISession>()
                 .Query<ContentItem, PublishLaterPartIndex>(index => index.ScheduledPublishUtc < _clock.UtcNow)
-                .With<ContentItemIndex>(index => index.Latest && !index.Published)
```

Comment:
For this one, two reasons.
- The ContentItemIndex is in the module directly. If it's necessary to check for latest/published then we usually add it to the new index.
- The index already has this contraint in place so there is no need to check. I am still doing in the code in case there could be some data corruption.

Reply:
Where does the index has this constraint in place? I've seen you added the same in `ScheduledPublishingBackgroundTask` but why there, why not here when creating the index?

Reply:
In the index:

```
if (contentItem.Published || !contentItem.Latest) { return null; }
```

With that there won't be any index record for a content item (version) that is published or not versioned. Even if an item becomes published, the record will then be removed, which is also why you don't have to delete any index record once a scheduled task is processed.

Reply:
PS: Some of our indexes have two records, one for the versioned and one for the published, in case we need to query them an just get one of these.

Reply:
Yeah but I think this is what happens in the original code too: There won't be an index for such items and there's no explicit index delete either.

However, it seems to me that what the added check in the bg task does now is filter out items where the publishing would do nothing. I don't think that warrants a warning log entry though, e.g. a user scheduling an item to be published but then publishing it by hand is not really a faulty operation.

Reply:
> a user scheduling an item to be published but then publishing it by hand

If that's the case the index should have removed the index record, that's my point, based on the code in the index.

Reply:
Correction: I think this is what happens too already. The index provider will return null in that case, and since the user doing the publish will be a document update it'll remove the index. I don't see how the second check in the bg task helps (unless somehow you modify the content item's flags directly from code.

Reply:
Yes! It shouldn't happen, and you had a filter in the query, which is as useless as this code but slower actually. And I didn't want to remove your check completely in case you really cared so I did the filtering in code and added a log too.

Reply:
I don't care :). So if two filters are not needed then let's just have a single one, in the filter provider.

Reply:
Done

Reply:
Great!